src/System.Windows.Forms/src/System/Windows/Forms/TreeNodeCollection.cs

```diff
@@ -66,6 +66,11 @@ internal int FixedIndex
                 TreeView tv = owner.treeView;
                 TreeNode actual = owner.children[index];
 
+                if (value is null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
```

Comment:
We want to validate before executing any other code:
```suggestion
                if (value is null)
                {
                    throw new ArgumentNullException(nameof(value));
                }

                TreeView tv = owner.treeView;
                TreeNode actual = owner.children[index];

```

Reply:
Done! Thanks for the review!